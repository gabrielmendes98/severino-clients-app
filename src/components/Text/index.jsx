import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Text as RNText } from 'react-native';
import styles from './style';

const Text = ({ children, style, size, color, weight, margin, ...props }) => {
  const customStyle = useMemo(
    () => styles({ size, color, weight, margin }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [color, size, weight],
  );

  return (
    <RNText style={[customStyle.text, style]} {...props}>
      {children}
    </RNText>
  );
};

Text.propTypes = {
  children: PropTypes.any.isRequired,
  color: PropTypes.string,
  margin: PropTypes.shape({
    top: PropTypes.number,
    right: PropTypes.number,
    bottom: PropTypes.number,
    left: PropTypes.number,
  }),
  size: PropTypes.number,
  style: PropTypes.object,
  weight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default Text;
