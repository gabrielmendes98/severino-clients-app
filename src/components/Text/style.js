import { StyleSheet } from 'react-native';
import { getTextColor, marginHandler } from 'common/styles/util';
import theme from 'common/styles/theme';

const styles = ({
  size = 1,
  color = 'main',
  weight = 'normal',
  margin,
  align,
}) =>
  StyleSheet.create({
    text: {
      color: getTextColor(color),
      fontSize: theme.fontSize(size),
      fontWeight: String(weight),
      textAlign: align,
      ...marginHandler(margin),
    },
  });

export default styles;
