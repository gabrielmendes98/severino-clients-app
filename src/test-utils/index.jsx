/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import MainProvider from 'common/contexts/Main';

const Wrapper = ({ children }) => <MainProvider>{children}</MainProvider>;

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
