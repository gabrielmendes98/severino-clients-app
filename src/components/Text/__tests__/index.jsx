import React from 'react';
import { rawRender } from 'test-utils';
import Text from '../index';

it('should render with correct styles with props', () => {
  const { toJSON } = rawRender(
    <Text
      color="light"
      margin={{ top: 2 }}
      size={1.5}
      weight="bold"
      align="center"
    >
      TEST TEXT
    </Text>,
  );

  expect(toJSON()).toMatchSnapshot();
});

it('should render with correct styles with default props', () => {
  const { toJSON } = rawRender(<Text>TEST TEXT</Text>);

  expect(toJSON()).toMatchSnapshot();
});
