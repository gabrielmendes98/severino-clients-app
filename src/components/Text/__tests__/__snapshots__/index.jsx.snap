// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with correct styles with default props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 14.4,
        "fontWeight": "normal",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "auto",
      },
      undefined,
    ]
  }
>
  TEST TEXT
</Text>
`;

exports[`should render with correct styles with props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#838383",
        "fontSize": 21.6,
        "fontWeight": "bold",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 16,
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  TEST TEXT
</Text>
`;
